cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -214,6 +223,7 @@ public void stop() {
     if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
       Closeables.closeQuietly(baseInjector.getInstance(ExploreClient.class));
     }
+    stopQuietly(startupChecker);
```

Comment:
This thread doesn't need to exist till the end of master right ? This can be stopped immediately after the checks are done.